informers.go

```diff
@@ -23,14 +34,89 @@ func getVirtCli() (kubecli.KubevirtClient, error) {
 	return virtCli, err
 }
 
-func getVmiInformer() (cache.SharedIndexInformer, error) {
-	virtCli, err := getVirtCli()
-	if err != nil {
-		return nil, err
-	}
-
+func getVmiInformer(virtCli kubecli.KubevirtClient) (cache.SharedIndexInformer, error) {
 	listWatcher := cache.NewListWatchFromClient(virtCli.RestClient(), "virtualmachineinstances", k8sv1.NamespaceAll, fields.Everything())
 	vmiInformer := cache.NewSharedIndexInformer(listWatcher, &k6tv1.VirtualMachineInstance{}, 1*time.Hour, cache.Indexers{})
+	
+	vmiInformer.AddEventHandler(cache.ResourceEventHandlerFuncs {
+		DeleteFunc: removeWarningLabel,
+	})
 
 	return vmiInformer, nil
 }
+
+func updateMachineTypes(virtCli kubecli.KubevirtClient) error {
+	vmList, err := virtCli.VirtualMachine(k8sv1.NamespaceAll).List(&k8sv1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, vm := range vmList.Items {
+		machineType := vm.Spec.Template.Spec.Domain.Machine.Type
+		machineTypeSubstrings := strings.Split(machineType, "-")
+		version := machineTypeSubstrings[2]
+		if len(machineTypeSubstrings) != 3 {
+			return nil
+		}
+		
+		if strings.Contains(version, "rhel") && version < "rhel9.0.0" {
+			machineTypeSubstrings[2] = latestVersion
+			machineType = strings.Join(machineTypeSubstrings, "-")
+			updateMachineType := fmt.Sprintf(`{"spec": {"template": {"spec": {"domain": {"machine": {"type": "%s"}}}}}}`, machineType)
+			
+			_, err = virtCli.VirtualMachine(vm.Namespace).Patch(vm.Name, types.StrategicMergePatchType, []byte(updateMachineType), &k8sv1.PatchOptions{})
+			if err != nil {
+				return err
+			}
+			
+			// add label to running VMs that a restart is required for change to take place
+			if vm.Status.Ready {
+				err = addWarningLabel(virtCli, &vm)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func addWarningLabel (virtCli kubecli.KubevirtClient, vm *k6tv1.VirtualMachine) error {
+	addLabel := fmt.Sprint(`{"metadata": {"labels": {"restart-vm-required": "true"}}}}}`)
+	_, err := virtCli.VirtualMachineInstance(vm.Namespace).Patch(vm.Name, types.StrategicMergePatchType, []byte(addLabel), &k8sv1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+	vmiList = append(vmiList, vm.Name)
+	
+	return nil
+}
+
+func removeWarningLabel(obj interface{}) {
+	vmi, ok := obj.(*k6tv1.VirtualMachineInstance)
+	if !ok {
+		return
+	}
+	
+	//check if deleted VMI is in list of VMIs that need to be restarted
+	vmiIndex := searchVMIList(vmi.Name)
+	if  vmiIndex == -1 {
+		return
+	}
+	
+	// remove deleted VMI from list
+	vmiList = append(vmiList[:vmiIndex], vmiList[vmiIndex+1:]...)
```

Comment:
Not that it's critical, but I think this could have been much easier with a map.

The map could be of type `vmisPendingUpdate := make(map[string]struct{})`.
Then, to search: `_, exists := vmisPendingUpdate[<vm>]`
To delete: `delete vmisPendingUpdate[vmisPendingUpdate, <vm>`